### PR TITLE
Add function `getMessages`

### PR DESCRIPTION
**Pull request by darkdef**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️/❌

### CODE REVIEW

File: src/MessageSource.php [hunk 1]
```diff
@@ -35,6 +35,11 @@ public function getMessage(string $id, string $category, string $locale, array $
         return dngettext($category, $id, $id, $n);
     }
 
+    public function getMessages(string $category, string $locale): array
+    {
+        throw new \RuntimeException('Function not allowed for this package.');
```
Comment (samdark) [inline src/MessageSource.php:40]: I think we need to implement that by using Yii 2 implementations:

- https://github.com/yiisoft/yii2/blob/master/framework/i18n/GettextMoFile.php
- https://github.com/yiisoft/yii2/blob/master/framework/i18n/GettextPoFile.php

Comment (samdark) [inline src/MessageSource.php:40]:
```suggestion
        throw new \RuntimeException('Unable to get all messages from gettext.');
```

Comment (samdark) [reply src/MessageSource.php:40]: But overall we'd better to provide a way to extract messages from gettext or make sure gettext can handle it.

Comment (samdark) [reply src/MessageSource.php:40]: @darkdef will disucss that with @Mister-42.

Comment (Mister-42) [reply src/MessageSource.php:40]: So far I indeed have been using gettext to handle extracting.

```bash
find src/ -name *.php | xargs xgettext --from-code=utf-8 --language=PHP --no-location --omit-header --sort-output --keyword=translate --output="locales/category.pot"
```

Out of laziness I just started using gettext in php, but the .pot files seems to be easy to convert to a PHP array if required.

Comment (samdark) [reply src/MessageSource.php:40]: OK. That should be put into docs.

**Review by samdark — APPROVED** [review]

File: src/MessageSource.php (hunk 1 above)
Comment (Mister-42) [reply src/MessageSource.php:40]: Note: gettext extraction is not aware of the Yii way of plurals. I am not sure how much this impacts etraction, as I am using PHP gettext natively

**Comment by Mister-42** [comment]
Can we get this in please? The package currently fatals out.
```
Class Yiisoft\Translator\Message\Gettext\MessageSource contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (Yiisoft\Translator\MessageReaderInterface::getMessages)
```

**Pull request merged**